cms/djangoapps/contentstore/views/component.py

```diff
@@ -47,17 +50,41 @@
     "basic-modal", "modal-button", "edit-xblock-modal",
     "editor-mode-button", "upload-dialog",
     "add-xblock-component", "add-xblock-component-button", "add-xblock-component-menu",
-    "add-xblock-component-menu-problem", "xblock-string-field-editor", "publish-xblock", "publish-history",
+    "add-xblock-component-support-legend", "add-xblock-component-support-level", "add-xblock-component-menu-problem",
+    "xblock-string-field-editor", "publish-xblock", "publish-history",
     "unit-outline", "container-message", "license-selector",
 ]
 
 
-def _advanced_component_types():
+def _advanced_component_types(show_unsupported):
     """
     Return advanced component types which can be created.
+
+    Args:
+        show_unsupported: if True, unsupported XBlocks may be included in the return value
+
+    Returns:
+        A dict of authorable XBlock types and their support levels (see XBlockStudioConfiguration). For example:
+        {
+            "done": "us",  # unsupported
+            "discussion: "fs"  # fully supported
+        }
+        Note that the support level will be "True" for all XBlocks if XBlockStudioConfigurationFlag
+        is not enabled.
     """
-    disabled_create_block_types = XBlockDisableConfig.disabled_create_block_types()
-    return [c_type for c_type in ADVANCED_COMPONENT_TYPES if c_type not in disabled_create_block_types]
+    enabled_block_types = _filter_disabled_blocks(ADVANCED_COMPONENT_TYPES)
+    if XBlockStudioConfigurationFlag.is_enabled():
```

Comment:
Note to reviewers: authorable_xblocks returns None if XBlockStudioConfigurationFlag.is_enabled() returns False. However, since there are other places where we explicitly check XBlockStudioConfigurationFlag.is_enabled(), I thought it was clearer to use that check here as well instead of comparing the return value of authorable_xblocks.